pkg/controller/controller_clusterservicebroker.go

```diff
@@ -751,23 +762,26 @@ func (c *controller) getCurrentServiceClassesAndPlansForBroker(broker *v1beta1.C
 	return existingServiceClasses.Items, existingServicePlans.Items, nil
 }
 
-func convertClusterServiceClassListToMap(list []v1beta1.ClusterServiceClass) map[string]*v1beta1.ClusterServiceClass {
+func convertAndFilterClusterServiceClassListToMap(list []v1beta1.ClusterServiceClass) map[string]*v1beta1.ClusterServiceClass {
 	ret := make(map[string]*v1beta1.ClusterServiceClass, len(list))
```

Comment:
not sure it matters much, but you could be allocating too much if the "if" in the for-loop below excludes a lot of classes.